id: python.django.correctness.model-save.django-db-model-save-super
patterns:
- pattern-inside: "class $MODEL(django.db.models.Model):\n    ...\n"
- pattern-not: "def save(self, ...):\n  ...\n  super($MODEL, self).save(...)\n"
- pattern-not: "def save(self, ...):\n  ...\n  super().save(...)\n"
- pattern: "def save(self, ...):\n  ...\n"
message: Detected a django model `$MODEL` is not calling super().save() inside of
  the save method.
languages:
- python
severity: WARNING
metadata:
  category: correctness
  technology:
  - django
  license: Commons Clause License Condition v1.0[LGPL-2.1-only]
  source: https://semgrep.dev/r/python.django.correctness.model-save.django-db-model-save-super
  shortlink: https://sg.run/6nN1
  semgrep.dev:
    rule:
      r_id: 9459
      rv_id: 110168
      rule_id: 9AU1BD
      version_id: nWTxPBL
      url: https://semgrep.dev/playground/r/nWTxPBL/python.django.correctness.model-save.django-db-model-save-super
      origin: community
